src/NUnitFramework/framework/Internal/Tests/TestSuite.cs

```diff
@@ -100,7 +100,15 @@ public void Sort()
         {
             if (!MaintainTestOrder)
             {
-                this.tests.Sort();
+                this.tests.Sort(delegate (ITest x, ITest y)
+                {
+                    var xKey = 0;
+                    var yKey = 0;
+                    if (x.Properties.ContainsKey("order")) int.TryParse(x.Properties["order"][0].ToString(), out xKey);
+                    if (y.Properties.ContainsKey("order")) int.TryParse(y.Properties["order"][0].ToString(), out yKey);
+                    return xKey.CompareTo(yKey) != 0
+                    ? xKey.CompareTo(yKey) : x.Name.CompareTo(y.Name); ;
```

Comment:
Regarding sort by name... we've been telling people that workaround is a
bad one and will not be supported in future for many years. However, let's
leave it for the moment. I still want to look at WorkItem ordering.


On Sun, Feb 14, 2016 at 6:05 PM, Joel Dickson <notifications@github.com>
wrote:

> In src/NUnitFramework/framework/Internal/Tests/TestSuite.cs
> <https://github.com/nunit/nunit/pull/1264#discussion_r52854216>:
>
> > @@ -100,7 +100,15 @@ public void Sort()
> >          {
> >              if (!MaintainTestOrder)
> >              {
> > -                this.tests.Sort();
> > +                this.tests.Sort(delegate (ITest x, ITest y)
> > +                {
> > +                    var xKey = 0;
> > +                    var yKey = 0;
> > +                    if (x.Properties.ContainsKey("order")) int.TryParse(x.Properties["order"][0].ToString(), out xKey);
> > +                    if (y.Properties.ContainsKey("order")) int.TryParse(y.Properties["order"][0].ToString(), out yKey);
> > +                    return xKey.CompareTo(yKey) != 0
> > +                    ? xKey.CompareTo(yKey) : x.Name.CompareTo(y.Name); ;
>
> My 2 cents.. I recommend this should be retained, because in the previous
> issue comments someone has listed a work around as using A_ B_ etc for the
> test name, so you may well have some customers out there that are using
> this, so if you change the order it may break someones test when they
> upgrade. But it's up to you mate :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nunit/nunit/pull/1264/files#r52854216>.
>